### PR TITLE
feat: check in goal frontend

### LINKED ISSUE #41: (Frontend) Check in goal
- Check in transition and animation for goal on Today
- Modal, clip-path

### PR DESCRIPTION
**Pull request by IamMrandrew**
- check in modal
- fixed some backend bugs on checkin
- changed cookie length

### CODE REVIEW

File: server/api/controllers/goal.js [hunk 1]
```diff
@@ -135,10 +135,11 @@ exports.check_in = (req, res, next) => {
     .then((user) => {
       user.onGoingGoals.forEach((element) => {
         if (element.goal_id == req.body.goal_id) {
-          element.check_in += req.body.check_in_time;
+          element.check_in = Number(req.body.check_in_time);
+          console.log(user, element.check_in);
           Goal.findById(element.goal_id).then((goal) => {
             if (element.check_in == goal.frequency) {
-              element.check_in = 0;
+              // element.check_in = 0;
```
Comment (tomas2050) [inline server/api/controllers/goal.js:141]: why commenting this line?

Comment (IamMrandrew) [reply server/api/controllers/goal.js:141]: Like if the goal is currently 0/2. I check in 2 times, the 次數will be 0/2 too.
But 應該係會變2/2, 然後到第二日變返0/2.  @tomas2050

Comment (tomas2050) [reply server/api/controllers/goal.js:141]: ok should be the same if we will delete check in at the end of the day

**Comment by tomas2050** [comment]
looks good on server side

**Review by tomas2050 — DISMISSED** [review]
looks good on server side!

**Review by janson0004 — APPROVED** [review]
LGTM!

**Review by janson0004 — APPROVED** [review]
LGTM

**Pull request merged**